app/models/deferred/visitor.rb

```diff
@@ -0,0 +1,4 @@
+module Deferred
+  class Visitor < ::Visitor
+  end
```

Comment:
Possibly, something as simple as removing the `.new` and `#initialize` methods might be enough - I'm sure that `Marshal.load` bypasses that.